### PR TITLE
backupccl: update backup detached tests with retry execution under transaction

### PR DESCRIPTION
**Pull request by Elliebababa**
update backup and restore detached tests with retry execution under transaction

Previously, backup and restore detached tests do not retry if transactions fail. They could be flaky tests and fail in CI check.

With this PR, backup and restore detached tests under transaction will not be aborted by retry errors.

Release notes: None.

### CODE REVIEW

**Review by pbardea — APPROVED** [review]
LGTM with a minor nit. Thanks!

**Review by pbardea — COMMENTED** [review]
Ah -- my comment from my previous review got swallowed.

File: pkg/ccl/backupccl/backup_test.go [hunk 1]
```diff
@@ -5059,39 +5059,39 @@ func TestDetachedBackup(t *testing.T) {
...
 	defer cleanupFn()
 
 	db := sqlDB.DB.(*gosql.DB)
 
 	// running backup under transaction requires DETACHED.
 	var jobID int64
-	tx, err := db.Begin()
-	require.NoError(t, err)
-	err = tx.QueryRow(`BACKUP DATABASE data TO $1`, LocalFoo).Scan(&jobID)
+	err := crdb.ExecuteTx(ctx, db, nil, func(tx *gosql.Tx) error {
```
Comment (pbardea) [inline pkg/ccl/backupccl/backup_test.go:5069]: Let's add a `/* opts */` comment after the `nil`, for readability to indicate what we're passing as `nil`. (Here and below.)

**Comment by Elliebababa** [comment]
TFTR!

bors r=pbardea

**Pull request merged**